python/GafferSceneUI/ShaderUI.py

```diff
@@ -600,7 +600,7 @@ def __parameters( self ) :
 	def __shaders( self ) :
 
 		if len( self ) > 0 :
-			uniqueShaders = { n.shaders()[ self[0] ].hash() : n.shaders()[ self[0] ] for n in self.__shaderNetworks if self[0] in n.shaders() }
+			uniqueShaders = { n.hash().append( self[0] ) : n.shaders()[ self[0] ] for n in self.__shaderNetworks if self[0] in n.shaders() }
```

Comment:
I'm not sure I'm convinced by this. In the case of two networks that are largely the same, it puts the "duplicate" icon on every shader, rather than on the few shaders that are different between networks. In the example below, browsing from ShaderTweaks1, `Noise` is now shown as a duplicate, even though it is identical in both networks. I think maybe the previous behaviour was more informative?

Reply:
While monkeying with shader networks to make sure the fix worked I was interpreting the duplicate icon as

"the shader `Noise` occurs in more than one network, so if you tweak it, the changes will happen on multiple networks"
instead of 
"the shader `Noise` occurs in multiple places and is not the same everywhere, so user beware".

That would be a behavior change though so I agree it's probably best to revert this line.